Issue 2423: Flag to enable mkdocs plugin

vijikarthi opened the pull request:
**Changelog description**
* Adds a flag to enable mkdocs plugin to generate site documentation.

**Purpose of the change**
Fixes #2423

**What the code does**
To support `flink-connector` build (https://github.com/pravega/flink-connectors/issues/112) where `pravega` is referenced as a git sub-module and the usage of `mkdocs` plugin is broken since the plugin is not able to find git configuration file (sub-modules don't keep the same configuration file as regular project) which is used to derive the `repoUri`.

This change adds a flag to enable the mkdocs plugin. If disabled, then the described problem won't manifest.

**How to verify it**
Build pravega and flink-connector project and make sure there is no error or side-effects due to this configuration

fpj commented:
@vijikarthi 

> It sets the publish.repoUri ext variable to use pravega git repo URL as default publish URL.

This sounds like a hack, is there anything better we can do to fix this?

vijikarthi commented:
> This sounds like a hack, is there anything better we can do to fix this?

@fpj I have removed the default `repoUri` configuration and added a conditional option (`-PenableMkdocs`) to enable `mkdocs` plugin.
